service/internal/cloudtags/spec.go

```diff
@@ -0,0 +1,10 @@
+package cloudtags
+
+import "context"
+
+type Interface interface {
+	// ClusterLabelsNotEqual compares current cluster labels with the stack tags passed
+	ClusterLabelsNotEqual(ctx context.Context, clusterID string, stags map[string]string) (bool, error)
```

Comment:
* It is rather unfortunate to negate the boolean with a function name. Something should rather return true so that you can simply do `!true` in the code if you have to. People are used to that and understand it more easily. Having `Not` in the same forces me to very carefully read and think. 
* The interface should be clean. The tags should not be given, but managed internally. 

```suggestion
	ClusterLabelsEqual(ctx context.Context, clusterID string) (bool, error)
```

Reply:
Indeed I change it because I thought `!` was easier to miss than a negative expression. But I can revert it easily :)